### ISSUE TITLE
undefined method `io` for StubSocket

### ISSUE THREAD

**Issue by alexrs** [bblimke/webmock#854]
### What
I'm working on a project using Rails, Webmock and VCR, and I came across this interesting issue:

```
NoMethodError: 
undefined method `io' for #<StubSocket:0x00007fa1fd1d94e8 @read_timeout=10>
```

### How to reproduce it
The steps to reproduce this issue are:
1. Run a test that executes this [line in `bundler`](https://github.com/bundler/bundler/blob/master/lib/bundler/vendored_persistent.rb#L36) to record a VCR cassete. The test passes and the cassete is recorded successfully.
2. Run the test again to use the cassete. The test will fail with the aforementioned error.

I have been digging into this, and [`StubSocket`](https://github.com/bblimke/webmock/blob/master/lib/webmock/http_lib_adapters/net_http.rb#L238) doesn't have an `io` method, but [`connection.instance_variable_get(:@socket)`](https://github.com/bundler/bundler/blob/master/lib/bundler/vendored_persistent.rb#L34) returns a [`BufferedIO`](https://github.com/ruby/ruby/blob/master/lib/net/protocol.rb#L113) object that defines `io`.

### tl;dr
When the test is run without a VCR cassete, we don't use webmock, but when the cassete is present, webmock is used, and therefore a `StubSocket` object is returned instead of a `BufferedIO` object.

**Comment by machty** [comment]
FYI i hackishly solved this by putting the following in `spec/support/stub_socket.rb`

```
class StubSocket
  def io
    OpenStruct.new
  end
end

```
